packages/cli/src/utils/getClientVersion.ts

```diff
@@ -33,6 +39,67 @@ async function getPrismaClientVersionFromNodeModules(cwd: string = process.cwd()
   }
 }
 
+type DriverAdapterVersion = {
+  name: OfficialDriverAdapters
+  version: string
+  underlyingDriver: {
+    name: OfficialUnderlyingDrivers
+    version: string | null
+  }
+}
+
+/**
+ * Try reading the Prisma Driver Adapter versions from the local package.json.
+ * This also includes the underlying driver versions, if found. They are looked up
+ * among the `dependencies` only, with the exception of `wrangler` for `@prisma/adapter-d1`,
+ * which can also be a `devDependency`.
+ * Only official, well-known Driver Adapters are considered.
+ */
+export async function getPrismaDriverAdapterVersionsFromLocalPackageJson(
+  cwd: string = process.cwd(),
+): Promise<Array<DriverAdapterVersion> | null> {
+  try {
+    const pkgJsonPath = await pkgUp({ cwd })
+
+    if (!pkgJsonPath) {
+      return null
+    }
+
+    const pkgJsonString = await fs.promises.readFile(pkgJsonPath, 'utf-8')
+    const pkgJson = JSON.parse(pkgJsonString) as {
+      dependencies?: Record<string, string>
+      devDependencies?: Record<string, string>
+    }
+
+    const adapterVersions = Object.entries((pkgJson.dependencies ?? {}) as Record<string, string>)
+      // Find all known Driver Adapter dependencies, usually `@prisma/adapter-*`:
+      .filter(([key]) => isOfficialDriverAdapter(key)) as Array<[OfficialDriverAdapters, string]>
+
+    const adapterVersionsWithUnderlyingDrivers = adapterVersions.map(([adapterName, adapterVersion]) => {
+      const underlyingDriver = underlyingDriverAdaptersMap[adapterName]
+      let underlyingDriverVersion = pkgJson.dependencies?.[underlyingDriver] ?? null
+
+      // `wrangler` is the only underlying driver that can be just a `devDependency`.
+      if (underlyingDriverVersion === null && underlyingDriver === 'wrangler') {
+        underlyingDriverVersion = pkgJson.devDependencies?.[underlyingDriver] ?? null
+      }
```

Comment:
We should look in `dependencies` too because people will sometimes put it there as well

Reply:
Right, then I would say the opposite 😄 

In general this makes sense.

But it's possible that people misplace the adapter packages and place them in `devDependencies`, and we should look in both places without assuming they would only be for sure in "devDependencies" or "dependencies".

Reply:
TLDR: we better not have any assumption on where the package is placed